### ISSUE TITLE
TCP KEEPALIVE does not work on MacOS

### ISSUE THREAD

**Issue by rclapp** [cubewise-code/tm1py#1072]
When enabling TCP Keep Alive on MacOS connection fails on line 476
```python

            socket_options = urllib3.connection.HTTPConnection.default_socket_options + [
                (socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1),
                (socket.IPPROTO_TCP, socket.TCP_KEEPIDLE, self.TCP_SOCKET_OPTIONS['TCP_KEEPIDLE']),
                (socket.IPPROTO_TCP, socket.TCP_KEEPINTVL, self.TCP_SOCKET_OPTIONS['TCP_KEEPINTVL']),
                (socket.IPPROTO_TCP, socket.TCP_KEEPCNT, self.TCP_SOCKET_OPTIONS['TCP_KEEPCNT'])]
```
Based on this documentation some socket options are only valid on windows.

https://docs.python.org/3.7/library/socket.html#:~:text=On%20Windows%2C%20TCP_KEEPIDLE%2C%20TCP_KEEPINTVL%20appear%20if%20run%2Dtime%20Windows%20supports.

**Comment by macsir** [comment]
Could you describe how you got into this issue? I tested it on MacOS and it worked fine.

**Comment by rclapp** [comment]
Funny you found that, I was actually testing v12 on OpenShift as a part of these tests. PA support was trying to figure out why timeouts were not being respected. I will share that link with IBM PA support as well.

**Comment by MariusWirtz** [comment]
solved with #1077
